src/get-started/flutter-for/ios-devs.md

```diff
@@ -966,264 +1584,379 @@ class _SampleAppPageState extends State<SampleAppPage> {
-```dart
-import 'dart:async' show Future;
-import 'package:flutter/services.dart' show rootBundle;
+class DemoApp extends StatelessWidget {
+  const DemoApp({super.key});
 
-Future<String> loadAsset() async {
-  return await rootBundle.loadString('my-assets/data.json');
+  @override
+  Widget build(BuildContext context) => const Scaffold(body: Signature());
 }
-```
-
-For images, Flutter follows a simple density-based format like iOS.
-Image assets might be `1.0x`, `2.0x`, `3.0x`, or any other multiplier.
-The so-called [`devicePixelRatio`][]
-expresses the ratio of physical pixels in a single logical pixel.
 
-Assets are located in any arbitrary folder&mdash;
-Flutter has no predefined folder structure.
-You declare the assets (with location) in
-the `pubspec.yaml` file, and Flutter picks them up.
+class Signature extends StatefulWidget {
+  const Signature({super.key});
 
-For example, to add an image called `my_icon.png` to your Flutter
-project, you might decide to store it in a folder arbitrarily called `images`.
-Place the base image (1.0x) in the `images` folder, and the
-other variants in sub-folders named after the appropriate ratio multiplier:
+  @override
+  State<Signature> createState() => SignatureState();
+}
 
-```
-images/my_icon.png       // Base: 1.0x image
-images/2.0x/my_icon.png  // 2.0x image
-images/3.0x/my_icon.png  // 3.0x image
-```
+class SignatureState extends State<Signature> {
+  List<Offset?> _points = <Offset?>[];
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onPanUpdate: (details) {
+        setState(() {
+          RenderBox? referenceBox = context.findRenderObject() as RenderBox;
+          Offset localPosition =
+              referenceBox.globalToLocal(details.globalPosition);
+          _points = List.from(_points)..add(localPosition);
+        });
+      },
+      onPanEnd: (details) => _points.add(null),
+      child:
+      CustomPaint(
+        painter: SignaturePainter(_points),
+        size: Size.infinite,
+      ),
+    );
+  }
+}
 
-Next, declare these images in the `pubspec.yaml` file:
+class SignaturePainter extends CustomPainter {
+  SignaturePainter(this.points);
 
-```yaml
-assets:
- - images/my_icon.png
-```
+  final List<Offset?> points;
 
-You can now access your images using `AssetImage`:
+  @override
+  void paint(Canvas canvas, Size size) {
+    final Paint paint = Paint()
+      ..color = Colors.black
+      ..strokeCap = StrokeCap.round
+      ..strokeWidth = 5.0;
+    for (int i = 0; i < points.length - 1; i++) {
+      if (points[i] != null && points[i + 1] != null) {
+        canvas.drawLine(points[i]!, points[i + 1]!, paint);
+      }
+    }
+  }
 
-<?code-excerpt "lib/images.dart (AssetImage)"?>
-```dart
-AssetImage('images/a_dot_burr.jpeg')
+  @override
+  bool shouldRepaint(SignaturePainter oldDelegate) =>
+      oldDelegate.points != points;
+}
 ```
 
-or directly in an `Image` widget:
+### Widget opacity
```

Comment:
Be careful of advising people to use the Opacity widget, especially in the case of an animation. No bueno. https://docs.flutter.dev/perf/best-practices#pitfalls

Reply:
This was an existing part of the document, I just changed where it is in the doc